validator/client/health_monitor.go

```diff
@@ -0,0 +1,123 @@
+package client
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v6/config/params"
+	"github.com/OffchainLabs/prysm/v6/validator/client/iface"
+	"github.com/sirupsen/logrus"
+)
+
+type healthMonitor struct {
+	ctx       context.Context
+	cancel    context.CancelFunc
+	v         iface.Validator
+	maxFails  int
+	healthyCh chan bool // emits true → healthy, false → unhealthy
+	fails     int
+	isHealthy bool
+	sync.RWMutex
+}
+
+// newHealthMonitor
+func newHealthMonitor(
+	parentCtx context.Context,
+	parentCancel context.CancelFunc,
+	maxFails int,
+	v iface.Validator,
+) *healthMonitor {
+	m := &healthMonitor{
+		ctx:       parentCtx,
+		cancel:    parentCancel,
+		maxFails:  maxFails,
+		v:         v,
+		healthyCh: make(chan bool, 1),
+	}
+
+	// Prime channel with the current status so consumers have an initial value.
```

Comment:
Why is it important that there is an initial value?

Reply:
The runner loop needs it now that the channel is healthy to run

Reply:
good point let me retry this I moved this over but thought i should set it up in the initialization let me try